network/nxos/nxos_config.py

```diff
@@ -0,0 +1,268 @@
+#!/usr/bin/python
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+DOCUMENTATION = """
+---
+module: nxos_config
+version_added: "2.1"
+author: "Peter sprygada (@privateip)"
+short_description: Manage Cisco NXOS configuration sections
+description:
+  - Cisco NXOS configurations use a simple block indent file sytanx
+    for segementing configuration into sections.  This module provides
+    an implementation for working with NXOS configuration sections in
+    a deterministic way.  This module works with either CLI or NXAPI
+    transports.
+extends_documentation_fragment: nxos
+options:
+  commands:
```

Comment:
ramblings: another thought would be to make it 'rule' so order can be ignored and single rules be passed to the module (but this can become inefficient ... squashing would help but then we are back to checking order).